src/core/QueryManager.ts

```diff
@@ -361,6 +367,28 @@ export class QueryManager<TStore> {
         query: mutation.document,
         variables: mutation.variables,
       });
+      if (isExecutionPatchResult(result) && isNonEmptyArray(result.incremental)) {
+        const diff = cache.diff<TData>({
+          id: "ROOT_MUTATION",
+          // The cache complains if passed a mutation where it expects a
+          // query, so we transform mutations and subscriptions to queries
+          // (only once, thanks to this.transformCache).
+          query: this.transform(mutation.document).asQuery,
+          variables: mutation.variables,
+          optimistic: false,
+          returnPartialData: true,
+        });
+        const mergedData = mergeIncrementalData<TData | undefined>(diff.result, result);
```

Comment:
```suggestion
        const mergedData = mergeIncrementalData(diff.result, result);
```
Since you've typed `cache.diff` above with `TData`, do you need to pass this generic? I think type inference will work here since it should be able to infer `TData` from the first argument.

Reply:
Resolved in https://github.com/apollographql/apollo-client/pull/10368/commits/31949b340454b176fba65b0f46d92afc23b3a735.